### DISCUSSION TITLE
Cant speed up with mpirun while using lammps

### DISCUSSION THREAD

**Discussion by PPdavidXD** [deepmodeling/deepmd-kit#540] [Q&A]
Hi
I'm running lammps with the dp potential on Ubuntu16.04. my CPU is amd 3960x and GPU is 2080ti
I tried to use openmpi and mpich to perform multicore simulation by running mpirun -np 4 lmp -in in.lmp
I tried reinstall my system and install deepmd with conda (which will install mpich3.3 by default) to make sure my system is clean and no conflict of openmpi and mpich.
I tried to use larger system up to 3000 atoms, however, using mpirun still didn't speed up the simulation. 

Could you give me some suggestion of where the problem is? Thank you

**Comment by njzjz** [comment]
Do you use cpus or gpus? GPU version + mpi will try to use multiple GPUs. Do you use the mpi installed by conda?

**Reply by PPdavidXD** [reply]
I use single cpu and gpu. The deepmd-kit is of GPU version. I think that should be the reason. The mpich is installed by conda.
Is there a way to use multi cpu cores and one gpu? Will it be faster to use the cpu version with multi cpu cores?
Thank you!

**Reply by njzjz** [reply]
You can try OpenMP instead of MPI. See https://github.com/deepmodeling/deepmd-kit/discussions/469#discussioncomment-563487 to control the number of threads.

**Reply by PPdavidXD** [reply]
Thank you very much.
One more question, I try to use deepmd with RTX3090 and I found that these new GPU are only supported with CUDA 11 version.
Is there any method to quick install with conda? or where can I download the runfile with CUDA 11.1?
Thank you again for your help!

**Reply by njzjz** [reply]
We only provided lammps built with CUDA 11.1 via conda but not python interface because of lack of upstream dependencies. For python interface, you may try `pip install tensorflow deepmd-kit`.

**Reply by PPdavidXD** [reply]
Really appreciate your help!
I tried install the c++ interface of TF2.4.1+CUDA11.1+dp1.3.3 with lammps and it works.
Thank you so much.
